handlers/smshandler/smshandler.go

```diff
@@ -0,0 +1,274 @@
+package smshandler
+
+import (
+	"encoding/binary"
+	"encoding/csv"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/nyaruka/phonenumbers"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/metadb"
+	"go.vocdoni.io/dvote/log"
+)
+
+// DefaultMaxSMSattempts defines the default maximum number of SMS allowed attempts.
+const DefaultMaxSMSattempts = 5
+
+// SmsHandler is a handler that requires a simple math operation to be resolved.
+type SmsHandler struct {
+	stg                 Storage
+	forceElectionsMatch bool
+	challengeDB         db.Database
+	challengeLock       sync.RWMutex
+	mathRandom          *rand.Rand
+	SendChallengeFunc   func(phone *phonenumbers.PhoneNumber, challenge int) error
+}
+
+// GetName returns the name of the handler
+func (sh *SmsHandler) GetName() string {
+	return "smsHandler"
+}
+
+// Init initializes the handler.
+// Takes one argument for persistent data directory.
+func (sh *SmsHandler) Init(opts ...string) error {
```

Comment:
why isn't the signature here `Init(dir string, maxAttempts int) error`?

Reply:
Because it implements the `handlers` interface: https://github.com/vocdoni/blind-csp/blob/master/handlers/handlers.go